context.go

```diff
@@ -85,12 +113,12 @@ func (context *Context) Run(addressesToData map[string]any, timeout time.Duratio
 
 	// Save the Go pointer references to addressesToData that were referenced by the encoder
 	// into C ddwaf_objects. libddwaf's API requires to keep this data for the lifetime of the ddwaf_context.
-	defer context.cgoRefs.append(encoder.cgoRefs)
+	defer context.cgoRefs.append(persistentEncoder.cgoRefs)
```

Comment:
What allows us not to append `ephemeralData.cgoRefs`?
IIUC, we can avoid it because the Go reference is kept by this function call.
But if that's the case, I would try to ensure it even more explicitly to avoid potential compiler optimizations somehow removing the reference. WDYT?

Reply:
I didn't find any test on ephermeral addresses and that might be why this problem wasn't caught.

Reply:
There *are* definitely tests on ephemeral addresses. Not in the concurrent suite (but I also am not sure how this would uncover anything the other test hasn't).